plugin/utils/compiler_builtins.py

```diff
@@ -0,0 +1,197 @@
+"""Get compiler built-in flags."""
+
+import logging as _logging
+
+_log = _logging.getLogger("ECC")
+
+
+class CompilerBuiltIns:
+    """
+    Get the built in flags used by a compiler.
+
+    This class tries to retrieve the built-in flags of a compiler.
+    As an input, it gets the call to a compiler plus some default
+    flags. It tries to guess some further required inputs and then
+    queries the compiler for its built-in defines and include paths.
+    """
+
+    __cache = dict()
+
+    __DEFINES_BLACKLIST = [
+        "__USER_LABEL_PREFIX__",
+        "__STDC_HOSTED__",
+        "__REGISTER_PREFIX__",
+        "__STDC__"
+    ]
+
+    def __init__(self, args):
+        """
+        Create an object holding the built-in flags of a compiler.
+
+        This constructs a new object which holds the built-in flags
+        used by a compiler. The `args` is the call to the compiler; either
+        a string or a list of strings. If a list of strings is provided, it
+        is interpreted as the call of a compiler (i.e. the first entry
+        is the compiler to call and everything else are arguments to the
+        compiler). If a single string is given, it is parsed into a string
+        list first.
+        """
+        from shlex import split
+        super().__init__()
+        self._defines = list()
+        self._include_paths = list()
+        if isinstance(args, str):
+            # Parse arguments into list of strings first
+            args = split(args)
+        # Guess the compiler and standard:
+        (compiler, std) = self._guess_compiler(args)
+        if compiler is not None:
+            # Guess the language (we need to pass it to the compiler
+            # explicitly):
+            language = self._guess_language(compiler)
+            # Get defines and include paths from the compier:
+            cfg = (compiler, std, language)
+            self._debug("Getting default flags for {}".format(cfg))
```

Comment:
Use log here?

Reply:
I added a `_debug` utility method which writes to the log (and adds a prefix so we quickly see the message originates from that class). Is this what you mean?